transmart-rest-api/grails-app/controllers/org/transmartproject/rest/QueryController.groovy

```diff
@@ -136,6 +139,27 @@ class QueryController extends AbstractQueryController {
 
     /**
      * Count endpoint:
+     * <code>/v2/observations/counts?constraint=${constraint}</code>
+     *
+     * Expects a {@link Constraint} parameter <code>constraint</code>.
+     *
+     * @return a the number of observations that satisfy the constraint and the number of associated patients.
+     */
+    def counts() {
+        def args = getGetOrPostParams()
+        checkForUnsupportedParams(args, ['constraint'])
+
+        Constraint constraint = bindConstraint(args.constraint)
+        if (constraint == null) {
+            return
```

Comment:
Why do we do this?

Reply:
On binding errors, `bindConstraint` will return null (and set response code 400).